### PR TITLE
Implement delegation in voting tools

### PR DESCRIPTION
**Pull request by dkijania**
I took this pleasure to implement delegation handling in voting tools as lack of this functionality caused some problems in tests.

Added one functional tests for verifications

### CODE REVIEW

File: src/voting-tools-rs/src/logic.rs [hunk 1]
```diff
@@ -122,15 +122,29 @@ impl Reg {
...
-                // TODO: this seems weird
+            Delegations::Delegated(delegations) => {
+                let mut metadata_list = MetadataList::new();
+                for (delegation, weight) in delegations {
+                    let mut inner_metadata_list = MetadataList::new();
+                    inner_metadata_list.add(
+                        &TransactionMetadatum::new_bytes(hex::decode(
+                            delegation.trim_start_matches("0x"),
+                        )?)
+                        .unwrap(),
```
Comment (cameron1024) [inline src/voting-tools-rs/src/logic.rs:133]: How confident are we that this panic is impossible to reach?

File: src/voting-tools-rs/src/testing/test_api/fake.rs [hunk 2]
```diff
@@ -63,7 +63,19 @@ impl DataProvider for MockDbProvider {
                             if let Ok(data) = metadata.as_bytes() {
                                 Delegations::Legacy(hex::encode(data))
                             } else {
-                                Delegations::Delegated(vec![])
+                                let mut delegations = vec![];
+                                let delgation_list = metadata.as_list().unwrap();
+                                for i in 0..delgation_list.len() {
```
Comment (cameron1024) [inline src/voting-tools-rs/src/testing/test_api/fake.rs:68]: Could we use `.enumerate()` here?

**Review by cameron1024 — APPROVED** [review]

File: src/voting-tools-rs/src/testing/test_api/fake.rs (hunk 2 above)
Comment (dkijania) [reply src/voting-tools-rs/src/testing/test_api/fake.rs:68]: I would love to but looks like MetadataList doesn't expose `.iter()`

File: src/voting-tools-rs/src/logic.rs (hunk 1 above)
Comment (dkijania) [reply src/voting-tools-rs/src/logic.rs:133]: Ok, adding error handling

**Pull request merged**